### PR TITLE
Obj-C Wrapper Updates

### LINKED ISSUE #896: Setting `ODWCommonDataContext` on `ODWPrivacyGuardInitConfig` crashes
**Describe your environment.** 
Crash is happening on iOS, both device and simulator.

**Steps to reproduce.**
This code is able to reproduce the crash on the last line:
```obj-c
let pgInitConfig = ODWPrivacyGuardInitConfig()
pgInitConfig.scanForUrls = false
pgInitConfig.useEventFieldPrefix = true
pgInitConfig.dataContext = ODWCommonDataContext()  //Crash here.
```

**What is the expected behavior?**
`pgInitConfig.dataContext` set correctly.

**What is the actual behavior?**
> App Name is obfuscated below

```
2021-07-04 19:12:47.256752+0530 #######App[3739:2747716] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[ODWCommonDataContext copyWithZone:]: unrecognized selector sent to instance 0x2811e8af0'
*** First throw call stack:
(0x197d6a754 0x1ac8317a8 0x197c6dc3c 0x197d6d2ac 0x197d6f5b0 0x1ac828c80 0x10838c0b8 0x109faa4dc 0x109faa71c 0x10987c5d0 0x109fff114 0x10987bacc 0x109886098 0x109885e4c 0x104886748 0x104d1dc2c 0x19a75dbf0 0x19a75e208 0x19a75ddd8 0x10c24c9f8 0x10488631c 0x105428804 0x197999cf8)
libc++abi: terminating with uncaught exception of type NSException
*** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[ODWCommonDataContext copyWithZone:]: unrecognized selector sent to instance 0x2811e8af0'
terminating with uncaught exception of type NSException
```

**Additional context.**
The issue potentially lies with `@property(readwrite, copy, nonatomic) ODWCommonDataContext* dataContext;`. This property should be `assign` instead of `readwrite`.

### LINKED ISSUE #897: [iOS] Setting PG_BaseType for the privacy guard events
**Scenario** 
We would want to set a custom base type for the privacy guard events. Currently the base type (PG_BaseType) is the same as the event name. 

This is working for Android when the `type` property is set for the `EventProperties`. But don't see a `type` property setter in iOS for ODWEventProperties

**SDK version**
iOS Version **3.5.148**

Doing this as done in Android:
```        // For privacy guard, set CFG_MAP_COMPAT's CFG_STR_COMPAT_PREFIX to "" to set our custom baseType for events.
        NSDictionary *compatLogConfig = @{TSOneDSConfigurationKey.cfgStrCompatPrefix: @""};
        [logConfiguration setValue:compatLogConfig forKey:TSOneDSConfigurationKey.cfgMapCompat];
```

### PR DESCRIPTION
**Pull request by sid-dahiya**
Fix Privacy Guard wrapper. Fixes #896 
Add SetType API for EventProperties. Closes #897

### CODE REVIEW

File: wrappers/obj-c/CMakeLists.txt [hunk 1]
```diff
@@ -29,20 +29,24 @@ endif()
...
+    ../../third_party/Reachability/
     )
 
 get_property(dirs DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR} PROPERTY INCLUDE_DIRECTORIES)
 foreach(dir ${dirs})
     message(STATUS "dir='${dir}'")
 endforeach()
 
-set(PLATFORM_FILES main.mm ODWLogManager.mm ODWLogger.mm ODWEventProperties.mm ODWLogConfiguration.mm ODWSemanticContext.mm)
+set(PLATFORM_FILES main.mm ODWLogManager.mm ODWLogger.mm ODWEventProperties.mm ODWLogConfiguration.mm ODWSemanticContext.mm ODWCommonDataContext.mm ODWPrivacyGuardInitConfig.mm ODWPrivacyGuard.mm ../../lib/pal/posix/NetworkInformationImpl.mm ../../third_party/Reachability/ODWReachability.m)
```
Comment (sid-dahiya) [inline wrappers/obj-c/CMakeLists.txt:49]: In one of the previous changes, this file was not updated correctly. As such, the Obj-C sample app was not being built.

**Comment by sid-dahiya** [comment]
@maxgolov is the Android failure what you're fixing with your PR?

File: wrappers/obj-c/CMakeLists.txt [hunk 2]
```diff
@@ -29,20 +29,24 @@ endif()
 set (PLATFORM_LIBS "")
 # Add flags for obtaining system UUID via IOKit
 if (CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-set (PLATFORM_LIBS "-framework Foundation -framework CoreFoundation -framework IOKit -framework Network")
+set (PLATFORM_LIBS "-framework Foundation -framework CoreFoundation -framework IOKit -framework Network -framework SystemConfiguration")
```
Comment (eduardo-camacho) [inline wrappers/obj-c/CMakeLists.txt:32]: Why do you need to add SystemConfiguration?
Since this is a framework, this would require adopters to also add such framework

File: wrappers/obj-c/ODWEventProperties.h [hunk 3]
```diff
@@ -95,6 +95,11 @@ typedef NS_ENUM(uint8_t, ODWDiagLevel)
 */
 @property(readonly, copy, nonatomic) NSDictionary<NSString*, NSNumber*> * piiTags;
 
+/*!
+ @brief Specify the Base Type of an event. This field is populated in Records.Type
```
Comment (eduardo-camacho) [inline wrappers/obj-c/ODWEventProperties.h:99]: nit: is seems brief was copied from line 262

File: wrappers/obj-c/main.mm [hunk 4]
```diff
@@ -17,29 +17,29 @@ int main(int argc, char** argv){
         ODWLogger* myLogger = [ODWLogManager loggerWithTenant: token];
 
         ODWPrivacyGuardInitConfig* pgInitConfig = [[ODWPrivacyGuardInitConfig alloc] init];
-        [[pgInitConfig useEventFieldPrefix]: TRUE]
+        pgInitConfig.useEventFieldPrefix = TRUE;
 
-        [pgInitConfig dataContext] = [[ODWCommondataContext alloc] init];
+        pgInitConfig.dataContext = [[ODWCommonDataContext alloc] init];
```
Comment (eduardo-camacho) [inline wrappers/obj-c/main.mm:22]: Question: why did you change all this assignments to dot notation?

**Review by eduardo-camacho — CHANGES_REQUESTED** [review]
Adding SystemConfiguration framework has implications on consumers, please confirm if you really need it

File: wrappers/obj-c/CMakeLists.txt (hunk 2 above)
Comment (sid-dahiya) [reply wrappers/obj-c/CMakeLists.txt:32]: This is being brought in due to third_party/Reachability code. Without this, the sample `main.mm` file fails with the linker errors below. Can you share what the recommendation here would be?
```
Consolidate compiler generated dependencies of target sample
[ 8%] Building CXX object CMakeFiles/sample.dir/main.mm.o
[ 16%] Building CXX object CMakeFiles/sample.dir/ODWLogManager.mm.o
[ 25%] Building CXX object CMakeFiles/sample.dir/ODWLogger.mm.o
[ 33%] Building CXX object CMakeFiles/sample.dir/ODWLogConfiguration.mm.o
[ 41%] Building CXX object CMakeFiles/sample.dir/ODWPrivacyGuardInitConfig.mm.o
[ 50%] Building CXX object CMakeFiles/sample.dir/ODWPrivacyGuard.mm.o
[ 58%] Linking CXX executable sample
Undefined symbols for architecture x86_64:
"_SCNetworkReachabilityCreateWithAddress", referenced from:
+[ODWReachability reachabilityWithAddress:] in ODWReachability.m.o
"_SCNetworkReachabilityCreateWithName", referenced from:
+[ODWReachability reachabilityWithHostname:] in ODWReachability.m.o
"_SCNetworkReachabilityGetFlags", referenced from:
-[ODWReachability isReachable] in ODWReachability.m.o
-[ODWReachability isReachableViaWiFi] in ODWReachability.m.o
-[ODWReachability connectionRequired] in ODWReachability.m.o
-[ODWReachability isConnectionOnDemand] in ODWReachability.m.o
-[ODWReachability isInterventionRequired] in ODWReachability.m.o
-[ODWReachability reachabilityFlags] in ODWReachability.m.o
"_SCNetworkReachabilitySetCallback", referenced from:
-[ODWReachability startNotifier] in ODWReachability.m.o
-[ODWReachability stopNotifier] in ODWReachability.m.o
"_SCNetworkReachabilitySetDispatchQueue", referenced from:
-[ODWReachability startNotifier] in ODWReachability.m.o
-[ODWReachability stopNotifier] in ODWReachability.m.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [sample] Error 1
make[1]: *** [CMakeFiles/sample.dir/all] Error 2
make: *** [all] Error 2
```

**Comment by sid-dahiya** [comment]
@eduardo-camacho was able to help me fix the actual issue in `main.mm`, which removed the need to update `CMakeLists.txt`.

**Review by eduardo-camacho — APPROVED** [review]

**Review by kindbe — APPROVED** [review]
LGTM
